Sources/ClientRuntime/Networking/Http/URLSession/FoundationStreamBridge.swift

```diff
@@ -110,32 +118,54 @@ class FoundationStreamBridge: NSObject, StreamDelegate {
         boundStreamBufferSize: Int? = nil,
         logger: LogAgent
     ) {
+        self.bridgeBufferSize = bridgeBufferSize
+        self.boundStreamBufferSize = boundStreamBufferSize ?? bridgeBufferSize
+        self.buffer = Data(capacity: bridgeBufferSize)
+        self.readableStream = readableStream
+        self.logger = logger
+        (inputStream, outputStream) = Self.makeStreams(boundStreamBufferSize: self.boundStreamBufferSize, queue: queue)
+    }
+
+    func replaceStreams(completion: @escaping (InputStream?) -> Void) async {
+        // Close the current output stream, since it will accept no more data and is about to
+        // be replaced.
+        await close()
+
+        // Replace the bound stream pair with new bound streams.
+        (inputStream, outputStream) = Self.makeStreams(boundStreamBufferSize: boundStreamBufferSize, queue: queue)
+
+        // Call the completion block.  When this method is called from `urlSession(_:task:needNewBodyStream:)`,
+        // the completion block will be that method's completion handler.
+        completion(inputStream)
+
+        // Re-open the `OutputStream` for writing.
+        await open()
+    }
+
+    private static func makeStreams(boundStreamBufferSize: Int, queue: DispatchQueue) -> (InputStream, OutputStream) {
```

Comment:
This method just extracts the stream creation logic from the initializer so it can be used to replace the streams as well as at this class's initialization.